app/controllers/application_controller.rb

```diff
@@ -841,12 +841,6 @@ def redirect_to_login
       # save original URL in a session so we can redirect after login
       session[:destination_url] = destination_url
       redirect_to path("/session/sso")
-    elsif SiteSetting.auth_immediately && !SiteSetting.enable_local_logins &&
-          Discourse.enabled_authenticators.length == 1 && !cookies[:authentication_data]
-      # Only one authentication provider, direct straight to it.
-      # If authentication_data is present, then we are halfway though registration. Don't redirect offsite
-      cookies[:destination_url] = destination_url
-      redirect_to path("/auth/#{Discourse.enabled_authenticators.first.name}")
```

Comment:
Why are we dropping this section?

Reply:
It was too aggressive with the redirects. Now that we use the routes, it caused it to always redirect directly to the external auth, never showing the login-required page

Reply:
Hmm, but that's what this setting is for, for going directly to the external auth, that's what the setting description says: 

![CleanShot 2025-01-24 at 11 27 52@2x](https://github.com/user-attachments/assets/339b80b1-ee67-4d87-80b4-722f47d5da01)

Reply:
I added a check for this `auth_immediately` setting to the frontend, so it should be ok now. It's not very clear to me why it was checking that setting, as it is more related to DiscourseConnect than anything else. We weren't doing any checks for this setting in any of the redirects in the frontend, even before any of my changes. I'll update the description of the setting to reflect the changes.